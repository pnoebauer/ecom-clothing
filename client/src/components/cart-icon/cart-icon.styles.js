import styled from 'styled-components/macro';

export const CartIconContainer = styled.div`
	width: 45px;
	height: 45px;
	position: relative;
	display: flex;
	justify-content: center;
	align-items: center;
	cursor: pointer;
`;

export const IconContainer = styled.div`
	height: 24px;
	width: 24px;
`;

export const ItemCountContainer = styled.span`
	position: absolute;
	font-size: 10px;
	font-weight: bold;
	bottom: 12px;
`;