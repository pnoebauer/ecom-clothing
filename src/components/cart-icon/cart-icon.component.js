import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import {
	CartIconContainer,
	IconContainer,
	ItemCountContainer
} from './cart-icon.styles';

import { ReactComponent as Icon } from '../../assets/shopping-bag.svg';
import { toggleCartHidden } from '../../redux/cart/cart.actions';
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';

const CartIcon = ({ toggleCart, totalItemQuantity }) => (
	<CartIconContainer onClick={toggleCart}>
		<IconContainer as={Icon} />
		<ItemCountContainer>{totalItemQuantity}</ItemCountContainer>
	</CartIconContainer>
);

const mapDispatchToProps = dispatch => {
	return {
		toggleCart: () => dispatch(toggleCartHidden())
	};
};

const mapStateToProps = createStructuredSelector({
	totalItemQuantity: selectCartItemsCount
});

export default connect(mapStateToProps,mapDispatchToProps)(CartIcon);