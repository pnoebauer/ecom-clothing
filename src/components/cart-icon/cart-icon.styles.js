import styled from 'styled-components';

export const CartIconContainer = styled.div`
	width: 45px;
	height: 45px;
	position: relative;
	display: flex;
	justify-content: center;
	align-items: center;
	cursor: pointer;
`;

export const IconContainer = styled.div`
	height: 24px;
	widows: 24px;
`;

export const ItemCountContainer = styled.span`
	position: absolute;
	font-size: 10px;
	font-weight: bold;
	bottom: 12px;
`;