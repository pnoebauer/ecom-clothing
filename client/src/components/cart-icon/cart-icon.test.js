import React from 'react';
import {shallow} from 'enzyme';
import {CartIcon} from './cart-icon.component';
import {CartIconContainer, ItemCountContainer} from './cart-icon.styles';

describe('CartIcon component', () => {
	let wrapper;
	let mockToggleCart;
	beforeEach(() => {
		mockToggleCart = jest.fn();

		const mockProps = {
			totalItemQuantity: 0,
			toggleCart: mockToggleCart,
		};

		wrapper = shallow(<CartIcon {...mockProps} />);
	});

	test('render CartIcon component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('call toggleCart when icon is clicked', () => {
		wrapper.find(CartIconContainer).simulate('click');
		expect(mockToggleCart).toHaveBeenCalled();
	});

	test('render the totalItemQuantity as the text', () => {
		const itemCount = parseInt(wrapper.find(ItemCountContainer).text());
		expect(itemCount).toBe(0);
	});
});
