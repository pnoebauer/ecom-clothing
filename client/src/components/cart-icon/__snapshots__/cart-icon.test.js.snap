// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component render CartIcon component 1`] = `
<cart-iconstyles__CartIconContainer
  onClick={[MockFunction]}
>
  <cart-iconstyles__IconContainer
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
  />
  <cart-iconstyles__ItemCountContainer>
    0
  </cart-iconstyles__ItemCountContainer>
</cart-iconstyles__CartIconContainer>
`;
